### PR TITLE
Include rooms in the list of candidates to share with

### PR DESCRIPTION
**Pull request by danxuliu**
Requires #10255 (this pull request should be rebased onto master once #10255  is merged)
Requires nextcloud/spreed#1050

### CODE REVIEW

File: apps/files_sharing/lib/Controller/DeletedShareAPIController.php [hunk 1]
```diff
@@ -165,4 +184,21 @@ public function undelete(string $id): DataResponse {
...
+	 *
+	 * @return OCA\Spreed\Share\Helper\Manager
+	 * @throws QueryException
+	 */
+	private function getRoomShareHelper() {
+		if (!$this->serverContainer->getAppManager()->isEnabledForUser('spreed')) {
+			throw new QueryException();
+		}
+
+		return $this->serverContainer->query(\OCA\Spreed\Share\Helper\DeletedShareAPIController::class);
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/DeletedShareAPIController.php]: meh

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 2]
```diff
@@ -978,4 +1019,21 @@ public function cleanup() {
...
+	 *
+	 * @return OCA\Spreed\Share\Helper\Manager
+	 * @throws QueryException
+	 */
+	private function getRoomShareHelper() {
+		if (!$this->serverContainer->getAppManager()->isEnabledForUser('spreed')) {
+			throw new QueryException();
+		}
+
+		return $this->serverContainer->query(\OCA\Spreed\Share\Helper\ShareAPIController::class);
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/ShareAPIController.php]: I dont like all this hardcoding :(

**Review by schiessle — APPROVED** [review]
tested and works as expected :+1:

**Comment by MorrisJobke** [comment]
@danxuliu @schiessle @nickvergessen 14 or 15?

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**